http/oidc/src/test/java/org/wildfly/security/http/oidc/OidcBaseTest.java

```diff
@@ -215,4 +231,18 @@ protected String getCookieString(HttpServerCookie cookie) {
         return header.toString();
     }
 
+    protected void checkForScopeClaims(Callback callback) throws InvalidJwtException {
+        Credential credential = ((IdentityCredentialCallback)callback).getCredential();
+        String token = ((BearerTokenCredential) credential).getToken();
+        JwtClaims jwtClaims = new JwtConsumerBuilder().setSkipSignatureVerification().setSkipAllValidators().build().processToClaims(token);
+        String scopes = jwtClaims.getClaimValueAsString("scope");
+        if (scopes != null) {
+            if (scopes.contains("email")) {
+                assertTrue(jwtClaims.getClaimValueAsString("email_verified").contains(String.valueOf(KeycloakConfiguration.EMAIL_VERIFIED)));
```

Comment:
It seems that Keycloak will set the `email` and `profile` scopes by default so just wondering if checking only `email_verified` and `profile` are actually enough to successfully test that the configured scopes have been set properly. As an example, I tried tweaking `testMultipleScopeValue` locally to only set the `openid` scope. I would have expected the test to fail in this case however it actually ended up passing. Is there a claim that we could check that would only be set if a specific scope other than the default ones has been configured?

Reply:
Looks like the `address` claim is a candidate for a claim that would confirm that the configured scopes is being used as expected.

Reply:
I don't think we can do address, because [org.keycloak.representations.idm.UserRepresentation](https://www.keycloak.org/docs-api/21.1.1/javadocs/org/keycloak/representations/idm/UserRepresentation.html#attributes)
does not have a way to set that. Plus when I added address as a scope, the request timed out. 
I saw this earlier with this API as well when I wanted to set `offline_access` as one of the scopes. It is kind of limiting in that sense.

Reply:
However, I can change the default list of scopes for the client to just OpenID, so that the results of those scopes are only included if the config asks for it.

Reply:
I tried adding `address` locally and was able to see the `address` added in the claims. So even if we can't actually set a value for it, we can still verify that it's present in the claims. This allows us to verify that setting the scope value actually resulted in the correct behaviour. With the test as is right now, we can't be sure that the setting the scope actually had an effect since the current claims will be present regardless of the current scope configuration.

Reply:
> However, I can change the default list of scopes for the client to just OpenID, so that the results of those scopes are only included if the config asks for it.

Yes another option would be to change the default list of scopes if that's easy to do. I quickly tried that yesterday but wasn't able to get things working as expected but maybe I was missing something.

Reply:
Yeah, I am trying that now and I am getting a class cast exception. 
I am looking more into it to see what is causing that.

Reply:
I was able to fix this. I added openid as the default scope and the others needed to be added as optional scopes.

Reply:
Thank you for this update! The tests look great now!

One last comment (that won't block merging either) is that right now we're checking if the scope from the JWT claims contains a certain value and then checking that value. Instead, it would be better to use the `expectedScope` String instead. It should be possible to do that by passing the `expectedScope` to `getCallbackHandler` instead of passing the boolean value.

Reply:
Fixed!